File: observations/Q1/run-2/graph.py
from typing import TypedDict

from langgraph.graph import END, START, StateGraph

from agent_topology.langgraph import describe
from agent_topology.spec import canonical_json


class State(TypedDict):
    message: str


def greet(state: State) -> dict[str, str]:
    return {"message": f"Hello, {state['message']}!"}


builder = StateGraph(State)
builder.add_node("greet", greet)
builder.add_edge(START, "greet")
builder.add_edge("greet", END)
graph = builder.compile()

if __name__ == "__main__":
    print(canonical_json(describe(graph)))

File: observations/Q1/run-2/graph.mjs
import { Annotation, END, START, StateGraph } from "@langchain/langgraph";
import { describe } from "@agent-topology/langgraph";
import { canonicalStringify } from "@agent-topology/spec";

const State = Annotation.Root({ message: Annotation() });
const graph = new StateGraph(State)
  .addNode("greet", (state) => ({ message: `Hello, ${state.message}!` }))
  .addEdge(START, "greet")
  .addEdge("greet", END)
  .compile();

const document = await describe(graph);
console.log(canonicalStringify(document));
